lib/galaxy/tools/parameters/basic.py

```diff
@@ -845,13 +843,12 @@ def get_legal_values(self, trans, other_values):
             return self.legal_values
 
     def from_json(self, value, trans, other_values={}):
+        if not value and not self.optional:
```

Comment:
I think the empty string should be an allowed value, so the test here should be `value is None` instead of `not value`.